boltdesignsystem/bolt
# DS-692 Animation Tool Set

## Jira

https://pegadigitalit.atlassian.net/browse/DS-692

## Summary

Animated classes with keyframes were added to `packages/global/styles/animations`.

## Details

There were created **fade**, **bounce**, **zoom** (**ins**/**outs**) combined with slide effects (**top**, **bottom**, **left**, **right**) + modifier classes controlling the cascading appearance of elements and delays options.

### fade
- fade in / out / in-up, out-up / in-down, out-down / in-left, out-left / in-right, out-right

### zoom
- zoom in / out / in-up, out-up / in-down, out-down / in-left, out-left / in-right, out-right

### bounce
- bounce in / out / in-up, out-up / in-down, out-down / in-left, out-left / in-right, out-right

---
Test pages were created in the `tests/animations` folder for experimenting.

## How to test

Check if animations work properly by adding specific classes to the elements considered to be animated. Make sure the performance of animating elements is sufficient.
